src/data/store/Tree.js

```diff
@@ -147,6 +147,15 @@ Ext.define('GeoExt.data.store.Tree', {
      */
     onLayerCollectionChanged: function(){
         var me = this;
+        // remove all filters as long as we take care of the changed collection
+        // but keep a reference so we can add them in later
+        var currentFilters = me.getFilters();
+        var restoreFilters = [];
+        currentFilters.each(function(currentFilter) {
+            restoreFilters.push(currentFilter);
+            me.removeFilter(currentFilter);
```

Comment:
Good comment, @weskamm. I didn't notice this.

It seems as if this isn't a problem, e.g. the following code

```javascript
var coll = Ext.create('Ext.util.MixedCollection');
        
coll.add(42);
coll.add(100);
coll.add(-100);
coll.add(17);
coll.add(Math.PI);

var iterated;

iterated = 0;
console.log('First iteration')
coll.each(function(entry) {
    iterated++;
    console.log('  … working on ' + entry);
    if (entry === -100) {
        console.log('  remove myself and last entry');
        coll.remove(entry);
        coll.removeAt(0);
        console.log('  add another entry');
        coll.add(4711)
    }
});
console.log('=> done. Iterated ' + iterated + ' times (expected 5)');
console.log('-----------------------------------------------------');

iterated = 0;
console.log('Second iteration')
coll.each(function(entry) {
    iterated++;
    console.log('  … working on ' + entry);
});
console.log('=> done. Iterated ' + iterated + ' times (expected 4)');
```

…produces this output…

```
First iteration
  … working on 42
  … working on 100
  … working on -100
  remove myself and last entry
  add another entry
  … working on 17
  … working on 3.141592653589793
=> done. Iterated 5 times (expected 5)
-----------------------------------------------------
 Second iteration
  … working on 100
  … working on 17
  … working on 3.141592653589793
  … working on 4711
=> done. Iterated 4 times (expected 4)
```

Do you think I need to change my code? I#ll gladly do it, it just seems not necessary.

But possibly it's a basic best-practice... :wink: